From 5b71364072b330240c00806ed73950a76500bab0 Mon Sep 17 00:00:00 2001
Message-Id: <5b71364072b330240c00806ed73950a76500bab0.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Shashank Sharma <shashank.sharma@intel.com>
Date: Mon, 26 Jan 2015 15:12:46 +0530
Subject: [PATCH 22/95] MUST_REBASE [VPG]: drm/i915: Wrapper function to get
 gmbus i2c adaptor

Adding wrapper function to get the gmbus i2c adaptor from the
i915 driver.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-5193
Change-Id: Ib7058b2deeb5f368b0342c1febd3c8e2eab754cf
Signed-off-by: Sharma, Shashank <shashank.sharma@intel.com>
Signed-off-by: Deepak M <m.deepak@intel.com>
---
 drivers/gpu/drm/i915/i915_adf_wrapper.c |    8 ++++++++
 include/drm/i915_adf.h                  |    1 +
 2 files changed, 9 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_adf_wrapper.c b/drivers/gpu/drm/i915/i915_adf_wrapper.c
index 81c2bed..8883d2b 100644
--- a/drivers/gpu/drm/i915/i915_adf_wrapper.c
+++ b/drivers/gpu/drm/i915/i915_adf_wrapper.c
@@ -88,6 +88,14 @@ struct pci_dev *i915_adf_get_pci_dev(void)
 }
 EXPORT_SYMBOL(i915_adf_get_pci_dev);
 
+struct i2c_adapter *intel_adf_get_gmbus_adapter(u8 port)
+{
+	/* port -1 to map pin pair to gmbus index */
+	return ((port >= GMBUS_PORT_SSC) && (port <= GMBUS_PORT_DPD)) ?
+		&i915_adf_dev->gmbus[port - 1].adapter : NULL;
+}
+EXPORT_SYMBOL(intel_adf_get_gmbus_adapter);
+
 void intel_adf_display_rpm_get(void)
 {
 	struct drm_i915_private *dev_priv;
diff --git a/include/drm/i915_adf.h b/include/drm/i915_adf.h
index 98aabc2..2709773 100644
--- a/include/drm/i915_adf.h
+++ b/include/drm/i915_adf.h
@@ -59,6 +59,7 @@ extern int i915_adf_driver_initialize(void);
 extern u8 intel_adf_get_platform_id(void);
 extern u16 intel_get_dsi_port_frm_vbt(void);
 extern void intel_get_vbt_disp_conf(void **child_dev, int *child_dev_num);
+extern struct i2c_adapter *intel_adf_get_gmbus_adapter(u8 port);
 #endif
 
 
-- 
1.7.9.5

